src/mscorlib/shared/System/Diagnostics/Tracing/TraceLogging/TraceLoggingMetadataCollector.cs

```diff
@@ -192,22 +192,44 @@ public void AddBinary(string name, TraceLoggingDataType type)
         }
 
         /// <summary>
-        /// Adds an array field to an event.
+        /// Adds a null-terminated string field to an event.
+        /// Compatible with core types: Utf16String, MbcsString.
+        /// Compatible with dataCollector method: AddNullTerminatedString(string).
         /// </summary>
         /// <param name="name">
         /// The name to use for the added field. This value must not be null.
         /// </param>
         /// <param name="type">
-        /// The type code for the added field. This must be a fixed-size type
-        /// or a string type. In the case of a string type, this adds an array
-        /// of characters, not an array of strings.
+        /// The type code for the added field. This must be a null-terminated string type.
         /// </param>
-        public void AddArray(string name, TraceLoggingDataType type)
+        public void AddNullTerminatedString(string name, TraceLoggingDataType type)
         {
             switch ((TraceLoggingDataType)((int)type & Statics.InTypeMask))
             {
                 case TraceLoggingDataType.Utf16String:
                 case TraceLoggingDataType.MbcsString:
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(type));
+            }
+
+            this.impl.AddNonscalar();
+            this.AddField(new FieldMetadata(name, type, this.Tags, this.BeginningBufferedArray));
```

Comment:
I can't say that this is correct or not.  Ideally Doug validates this.

Reply:
Thanks @vancem.  I am working with Doug offline to review this change.  I also am investigating the CI failure which looks like a real issue.